### PR TITLE
Add circleci yaml for Dockerfile test build

### PR DESCRIPTION
**Pull request by sdurrheimer**
Build example : https://circleci.com/gh/sdurrheimer/prometheus/4

@fabxc @juliusv

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ Prometheus' main distinguishing features as compared to other monitoring systems
 
 ## Architecture overview
 
-![](https://cdn.rawgit.com/prometheus/prometheus/62b69b0/documentation/images/architecture.svg)
+![](https://cdn.rawgit.com/prometheus/prometheus/62b69b0/documentation/images/architecture.svg) [![Circle CI](https://circleci.com/gh/prometheus/prometheus/tree/master.svg?style=svg)](https://circleci.com/gh/prometheus/prometheus/tree/master)
```
Comment (fabxc) [inline README.md]: Shouldn't this be next to the travis one?

Comment (sdurrheimer) [inline README.md]: yes, my bad, wrong line

**Comment by juliusv** [comment]
Nice. I just noticed that CircleCI signup requires access even to private repos, without ability to switch that off. That could be problematic for me. How does it manage access in general? If I create a separate GH user that is has the appropriate rights in the `prometheus` GitHub org, will it be able to get all the appropriate access rights in CircleCI automatically?

**Comment by sdurrheimer** [comment]
Hum I don't think so. It work with sources and forks repos, but I don't think you can build it from a collaborator account. 

> GitHub authorization
To run your tests, we need to check out your code from GitHub. When you sign up for Circle, you tell GitHub that you are authorizing us to check out your private repositories. You may revoke this permission at any time through your GitHub application settings page and by removing Circle's Deploy Keys and Service Hooks from your repositories' Admin pages.

>While Circle allows you to selectively test your projects, note that GitHub's permissions model is "all or nothing" — Circle gets permission to access all of a user's repositories or none of them. We have asked GitHub to provide finer-grained permissions, but they have indicated it will not be completed in the short term. Please contact GitHub if this is important to you.

source : https://circleci.com/security

**Comment by sdurrheimer** [comment]
I don't know much about organization permissions, it might work.
Only way to know is to test it. :)

**Comment by sdurrheimer** [comment]
Currently looking for a alternative : shippable, drone.io...

Seems like CircleCI is the only one to handle `docker build` natively.

**Comment by juliusv** [comment]
Ok, no worries. Should I just merge this then and see if it works? If we ever need to access it, we can still make separate accounts that don't have access to private repos.

**Comment by sdurrheimer** [comment]
It should work. You just need to create a prometheus CircleCI account and add this project to the build list.

**Comment by juliusv** [comment]
I'm just gonna merge this now and see what happens :)

**Pull request merged**